File: src/main/java/fr/sncf/osrd/infra/graph/EdgeDirection.java
package fr.sncf.osrd.infra.graph;

/** Encodes a direction along an edge. */
public enum EdgeDirection {
    START_TO_STOP(0),
    STOP_TO_START(1);

    public final int id;

    EdgeDirection(int id) {
        this.id = id;
    }

    /**
     * Gets the opposite of this direction
     * @return this opposite of this direction
     */
    public EdgeDirection opposite() {
        if (this == START_TO_STOP)
            return STOP_TO_START;
        return START_TO_STOP;
    }
}

File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/common/graph/EdgeDirection.kt
package fr.sncf.osrd.railjson.schema.common.graph

/** Encodes a direction along an edge. */
enum class EdgeDirection(val id: Int) {
    START_TO_STOP(0),
    STOP_TO_START(1);

    /**
     * Gets the opposite of this direction
     *
     * @return this opposite of this direction
     */
    fun opposite(): EdgeDirection {
        if (this == START_TO_STOP) return STOP_TO_START
        return START_TO_STOP
    }
}
